M2/Macaulay2/tests/CMakeLists.txt

```diff
@@ -0,0 +1,218 @@
+###############################################################################
+## This directory contains the rules for adding various tests to CTest.
+##
+## - list tests:      ctest -N
+## - build and test:  ctest --build-and-test
+## - rerun matching tests:
+##     ctest -R schorder --rerun-failed -V
+
+# TODO: some targets have different resource limits
+set(SUBDIRS normal slow gigantic goals quarantine engine threads)
+
+add_custom_target(M2-tests)
+add_dependencies(M2-tests M2-tests-ComputationsBook) # M2-tests-rationality
+
+###############################################################################
+## Default resource limits
+# TODO: add memory, virtual memory, and stack size
+# the tests aren't run through bash, so ulimit wouldn't work
```

Comment:
I don't think ulimits are the best practice for this purpose. For testing in particular, it's better to measure the memory usage and report the amount in addition to a red flag if it goes beyond a certain amount rather than interrupting the process right away.

Reply:
Why is it better?

How would you measure it?

Reply:
> Why is it better?

Because the ulimits don't warn you if after a change `help sum` suddenly takes 399MB of ram, even though that's definitely a red flag.

> How would you measure it?

I haven't put it all together yet, and it could be a project for someone else, but perhaps using Valgrind's [massif](https://valgrind.org/docs/manual/ms-manual.html#ms-manual.theoutputgraph)